### PR TITLE
DOC: fix runtests guidelines.

### LINKED ISSUE #13440: python runtest.py -t path-to-test.py failed
My issue is about running specific test in my development environnement.

#### Reproducing code example:

I use the command described in the [doc](http://scipy.github.io/devdocs/dev/contributor/runtests.html#runtests) : 
`python runtests.py -v -t scipy/optimize/tests/test_linprog.py::test_unknown_solver --no-build`

#### Error message:

```
=================================================================== test session starts ====================================================================
platform linux -- Python 3.8.5, pytest-6.2.2, py-1.10.0, pluggy-0.13.1 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /workspaces/scipy, configfile: pytest.ini
collected 0 items                                                                                                                                          

================================================================== no tests ran in 0.09s ===================================================================
ERROR: module or package not found: scipy/optimize/tests/test_linprog.py::test_unknown_solver (missing __init__.py?)
```

#### Scipy/Numpy/Python version information:
```
>>> import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info)
1.7.0.dev0+a274d79 1.19.5 sys.version_info(major=3, minor=8, micro=5, releaselevel='final', serial=0)
```

#### OS:
Working with docker with the following Dockerfile:
```Dockerfile
# ubuntu focal has python 3.8 as default
FROM ubuntu:focal

# add deadsnakes ppa to install python 3.7
RUN apt-get update && \
    apt-get install -y software-properties-common && \
    add-apt-repository ppa:deadsnakes/ppa

RUN apt-get update && apt-get install -y \
	python3.7 \
	python3.7-dev \
	python3-pip git \
	build-essential \
	vim \
	libatlas-base-dev \
	gfortran \
	libgfortran4 \
	liblapack-dev \
	curl \
	libgmp-dev \
	libmpfr-dev \
	libsuitesparse-dev \
	libmpc-dev

# setup pips, pip3.7 and pip3.8
RUN curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py && python3.7 get-pip.py && python3.8 get-pip.py && rm get-pip.py
```

### PR DESCRIPTION
**Pull request by V0lantis**
#### Reference issue
Fixes #13440.

#### What does this implement/fix?

This changes runtests guidelines.  The argument to the -t option should be a Python name (i.e. a dotted Python module path), not a file path.

### CODE REVIEW

**Comment by mdhaber** [comment]
To retrigger CI. I think an unrelated test was failing on master when this built.

File: doc/source/dev/contributor/runtests.rst [hunk 1]
```diff
@@ -29,33 +29,54 @@ To run tests on a particular submodule, such as ``optimize``, use the
...
+To run a particular test module, use the ``--test`` option:
 
 ::
 
-   python runtests.py -v -t scipy/optimize/tests/test_linprog.py
+   python runtests.py -v -t scipy.<module>.tests.<test_file>
 
-To run a test class, such as ``TestLinprogIPDense`` from
-``test_linprog.py``:
+Exemple for |test-linprog|_ file tests, run:
```
Comment (tylerjereddy) [inline doc/source/dev/contributor/runtests.rst:38]: typo--used French instead of English spelling of "Example"

**Review by tylerjereddy — COMMENTED** [review]
Based on the discussion in the matching issue, using the Python-name approach as per the changes here does seem a bit less likely to fail since it should work for both in-place builds and tests of installed builds without having to specify long absolute paths to build dir modules.

File: doc/source/dev/contributor/runtests.rst (hunk 1 above)
Comment (V0lantis) [reply doc/source/dev/contributor/runtests.rst]: Thank you for your review. Fix in db34472146a1e7a35a9c3a02bd3dc259acb766c0

**Comment by tylerjereddy** [comment]
Seems a bit more resilient to the mode of SciPy build/installation, and CI is green, so let's try this doc update.

**Comment by tylerjereddy** [comment]
Thanks @V0lantis

**Pull request merged**